docs/guidance/head-command.md

```diff
@@ -0,0 +1,146 @@
+
+    # wait for ray cluster to finish initialization
+    while true; do
+        ray health-check 2>/dev/null
+        if [ "$?" = "0" ]; then
+            break
+        else
+            echo "INFO: waiting for ray head to start"
+            sleep 1
+        fi
+    done
+
+    # Print the resources in the ray cluster after the cluster is ready.
+    python -c "import ray; ray.init(); print(ray.cluster_resources())"
+
+    echo "INFO: Print Ray cluster resources"
+```
+* We execute the script `ray_cluster_resources.sh` via the postStart hook. Based on [this document](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#container-hooks), there is no guarantee that the hook will execute before the container ENTRYPOINT. Hence, we need to wait for RayCluster to finish initialization in `ray_cluster_resources.sh`.
+
+* Example
+    ```sh
+    # Path: kuberay/
+    kubectl apply -f ray-operator/config/samples/ray-cluster.head-command.yaml
+
+    # Check ${RAYCLUSTER_HEAD_POD}
+    kubectl get pod -l ray.io/node-type=head
+
+    # Forward the port of Dashboard
+    kubectl port-forward --address 0.0.0.0 ${RAYCLUSTER_HEAD_POD} 8265:8265
+
+    # Open the browser and check the Dashboard (${YOUR_IP}:8265/#/job).
+    # You shold see a SUCCEEDED job with the following Entrypoint:
+    # 
+    # `python -c "import ray; ray.init(); print(ray.cluster_resources())"`
+    ```
+
+### Solution 2: Container command
+As we mentioned in the section "Timing 1: Before `ray start`", user-specified command will be executed before the `ray start` command. Hence, we can execute the `ray_cluster_resources.sh` in background by updating `headGroupSpec.template.spec.containers.0.args` in `ray-cluster.head-command.yaml`.
+
+```yaml
+# Parentheses for the command is required.
+command: ["(/home/ray/samples/ray_cluster_resources.sh&)"]
+```
+
+* Example
+    ```sh
+    # Path: kuberay/
+    # Update `command` to ["(/home/ray/samples/ray_cluster_resources.sh&)"] and comment out `postStart`.
+    kubectl apply -f ray-operator/config/samples/ray-cluster.head-command.yaml
+
+    # Check ${RAYCLUSTER_HEAD_POD}
+    kubectl get pod -l ray.io/node-type=head
+
+    # Check the logs
+    kubectl logs ${RAYCLUSTER_HEAD_POD}
+
+    # INFO: waiting for ray head to start
+    # .
+    # . => Cluster initialization
+    # .
+    # 2023-02-16 18:44:43,724 INFO worker.py:1231 -- Using address 127.0.0.1:6379 set in the environment variable RAY_ADDRESS
+    # 2023-02-16 18:44:43,724 INFO worker.py:1352 -- Connecting to existing Ray cluster at address: 10.244.0.26:6379...
+    # 2023-02-16 18:44:43,735 INFO worker.py:1535 -- Connected to Ray cluster. View the dashboard at http://10.244.0.26:8265
+    # {'object_store_memory': 539679129.0, 'node:10.244.0.26': 1.0, 'CPU': 1.0, 'memory': 2147483648.0}
+    # INFO: Print Ray cluster resources
+    ```
+    * The main difference between these two solutions is users can check the logs via `kubectl logs` with Solution 2.
```

Comment:
Can we put the pros and cons of either approach at the beginning of this section?  Also, since only one pro is listed, it sounds like Solution 2 is strictly better.  Is there any reason to use Solution 1? 

If we have a clear recommendation we could put (Recommended) for one of them to reduce the decisions the user has to make.

Reply:
Updated. https://github.com/ray-project/kuberay/pull/912/commits/9f63c636f944ad10f5d053777119598bedd3e95a